### DISCUSSION TITLE
Data card advanced "slot footer"

### DISCUSSION THREAD

**Discussion by DiegodeMiguel** [Telefonica/mistica-design#1710] [Proposals]
### Goal

We are looking to create an advanced card to include into CMS. We need it for O2 ES facelift where we need to include "O2 cloud" service with the mobile line information as first level into "Inicio tab".

What do you think?

### Context

We need to drop the facelift for O2 ES into June and the OB have included a new service for each mobile line, O2 cloud. I include the figma link about the new "Start tab" for O2 ES without O2 cloud and also with this.

### Impact

This proposal will affect to app builder to included it into the CMS for modular pages. So we need to create a "mistica-extended" component for this new use case for the actual "Data card" component.

### Discussion Questions

What do you think?
Do you have in mind some other component that can include this service at same level than the mobile line?
We think that its better to make a "mistica-extended" component than a slot that include "data progreess+footer". Are you agree?

### Do you have a visual proposal?

https://www.figma.com/file/KLtJiXB2fEDb9oW7WnU0vt/O2-Cloud?type=design&node-id=43-54389&mode=design&t=Kw7OofWKbzyIGlWk-4

**Comment by MontseTelefonica** [comment]
Hello! Do we have any updates on this?

**Reply by MontseTelefonica** [reply]
Is Mistica doing to add a footer to the data card?
FYI @sandrafernandezsimon

**Comment by aweell** [comment]
Buenas @DiegodeMiguel, @MontseTelefonica 

El principal problema de que sea ese slot de "data progreess+footer" es el siguiente:

* Por un lado se quiere ver el detalle de la linea, esto sería a través del onPress de la card
* Por otro lado esa parte del footer tiene el link para activar O2 Cloud

Lo que va a ocurrir es que el onPress afecta al contenido de toda la card por lo que todo el footer menos el area que ocupa el button link también te va a llevar al detalle de la linea. Lo que entendemos que no es lo deseado.

[Aquí un ejemplo](https://tinyurl.com/2ydpjbjg)

Por otro lado, antes de recurrir a construir un extended, vamos a explorar otras vías de encajar esta necesidad.

### Posibles soluciones

Hay varios escenarios posibles aquí:

1. Incluimos el footer como algo core
2. Otras soluciones alternativas revisadas offline
3. Que sea un extended 

#### 1. Incluirlo como core

Aquí surgen las mismas dudas que en su momento aparecieron con la advanced data card. Nuestra hipótesis es que no queda claro que con la doble interacción el onPress de una card pasa a ocupar únicamente la parte superior hasta el divider y si es evidente para el usuario que es una zona interactiva y no informativa.

Para desbloquear esta hipótesis lo adecuado sería realizar un test que no arroje luz sobre si es una buena práctica a incluir en el sistema.

Dejando la hipótesis a un lado, tenemos una idea más o menos clara de cómo se plantearía esta solución:

![Screenshot 2024-04-02 at 13 57 54](https://github.com/Telefonica/mistica-design/assets/44420072/76d7bfcd-4f2a-4ee4-ac06-edf329f0f50b)

#### 2. Otras soluciones alternativas revisadas offline

Listo aquí una serie de soluciones que se han planteado como factibles a día de hoy que podrían ser una alternativa al desarrollo de una nueva funcionalidad en el componente core:

* Se utiliza el "data progreess+footer" pero el detalle de la línea se hace explicito a través de un link
* Se utiliza una boxed row asociada a cada línea
* Se utiliza una boxed row general para todas las líneas que permite posteriormente decidir a que linea aplica
* Se utiliza un button secondary con la opción de activar el servicio en la propia card
* Se utiliza un button secondary con la opción de activar el sevicio y posteriormente decidir a que linea aplica
* En la línea de la opción anterior, pero en vez de usando un button, una quick action

Aquí el link a todas las soluciones: https://www.figma.com/file/KLtJiXB2fEDb9oW7WnU0vt/O2-Cloud?type=design&node-id=156%3A29748&mode=design&t=lYBOx1VdHieZuh9H-1

**Reply by MontseTelefonica** [reply]
Personalmente creo que lo que mejor encajaría es una de estas dos:
-Habilitar el footer en la data card 
-Usar un elemento ajeno a la card que te saque una distributiva de linea en caso de que haya más de una, ya que me preocupa ubicar un elemento sobre alamacenamiento sobre una card de consumo.

**Comment by yceballost** [comment]
Blocked until get research information

**Comment by MontseTelefonica** [comment]
Hola @yceballost. vuelvo con este hilo ya que es un requerimiento para O2 bastante urgente. ¿Se ha avanzado en este research? ¡Es mejor proceder con un slot para soportar este caso?
Gracias de antemano

**Reply by yceballost** [reply]
yo esto lo dejaría como extended hasta que podamos resolver la forma en hacer esto core

Idealmente estaría bien evolucionar las card para soportar el footer y tratar de eliminar la advanced data card de community, pero de momento esto está bastante fuera de nuestro scope

**Reply by MontseTelefonica** [reply]
Perfecto! Gracias

**Comment by yceballost** [comment]
Se cubrirá en el proyecto Cards Evolution que trabajaremos en Q2 https://github.com/Telefonica/mistica-design/issues/2106
